operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -25,6 +25,9 @@ func ExtractSecret(s *corev1.Secret, secretType lokiv1.ObjectStorageSecretType)
 		storageOpts.S3, err = extractS3ConfigSecret(s)
 	case lokiv1.ObjectStorageSecretSwift:
 		storageOpts.Swift, err = extractSwiftConfigSecret(s)
+	case lokiv1.ObjectStorageSecretS3VirtualHosted:
+		storageOpts.S3VirtualHosted, err = extractS3ConfigSecret(s)
+		storageOpts.SharedStore = lokiv1.ObjectStorageSecretS3
```

Comment:
IMHO I liked the previous mapping of the `s3-virtual-hosted` type into `s3` because it reuses the config rendering below.

Reply:
Do you mean revert to previous commit？

Reply:
Yes please

Reply:
I'm happy to do this!